src/operators/processing/raster/rgba_composite.cl

```diff
@@ -0,0 +1,45 @@
+__kernel void rgba_composite_kernel(__constant const IN_TYPE0 *red_data, __constant const RasterInfo *red_info,
+                                    __constant const IN_TYPE1 *green_data, __constant const RasterInfo *green_info,
+                                    __constant const IN_TYPE2 *blue_data, __constant const RasterInfo *blue_info,
+                                    __global uint *rgba_data, __constant const RasterInfo *rgba_info,
+                                    double red_min, double red_max, double red_scale,
+                                    double green_min, double green_max, double green_scale,
+                                    double blue_min, double blue_max, double blue_scale) {
+    // determine sizes from first raster
+    const int row_width = red_info->size[0];
+    const int number_of_pixels = red_info->size[0] * red_info->size[1] * red_info->size[2];
+
+    const size_t pixel_index = get_global_id(0) + get_global_id(1) * row_width;
+    if (pixel_index >= number_of_pixels) return;
+
+    const bool is_any_nodata = ISNODATA0(red_data[pixel_index], red_info) ||
+                               ISNODATA1(green_data[pixel_index], green_info) ||
+                               ISNODATA2(blue_data[pixel_index], blue_info);
+
+    // initialize `rgba_data` with alpha value, depending on if there is any NODATA
+    rgba_data[pixel_index] = is_any_nodata ? 0 : (255 << 24);
```

Comment:
generate the pixel value in a seperate variable and write this to output. This will use a register or local memory and helps debuging

Reply:
see refactoring in https://github.com/umr-dbs/mapping-core/pull/26/commits/4b0b7af37babd13ea8423fafdb2dbf35ed343724